### DISCUSSION TITLE
Adding more factors to my model increases the SHAP values. Does it mean that the model gets better?

### DISCUSSION THREAD

**Discussion by sibeleker** [shap/shap#1509] [Q&A]
I am trying to understand the rank-order of feature importance of a set of factors using XGBoost+SHAP combination. My first model has 4 features, and in a second model I add 2 more features. The absolute shapley values significantly increase in the second model. Does this indicate an increase in the explanatory power of the model, like higher R2 values do in a regression model?

Thanks in advance :)

**Comment by barnardp** [comment]
Hi Sibeleker,
I don't think that an increase in the absolute values of your shapley values necessarily implies that the explanatory power of your model increases. Keep in mind that the conventional shapley values (by conventional I mean those computed based on the model prediction) are effectively a decomposition of the model's output amongst the features of the model, and that most importantly, when you sum up the shapley values plus the baseline value, the total should be the same as the output value which you are trying to explain. 
In other words, _I think_ the fact that the magnitude of your shapley values are higher in the second model is probably due to the fact that the average prediction of you second model has increased in magnitude in comparison to your first model (i.e. higher variance) - but this doesn't necessarily imply that the predictions are more accurate than the first model. I have seen some papers though that show how one could compute R2 equivalents using the conventional shapley values which I think might be what you're looking for in this case. For example please see the following paper: https://arxiv.org/abs/1908.09718

Kindest regards, 
Pieter
